lib/python/pyflyby/_comms.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import absolute_import, division
+
+from   pyflyby._log             import logger
+import six
+
+# These are comm targets that the frontend (lab/notebook) is expected to
+# register. At this point, we handle only missing imports and
+# formatting imports
+
+MISSING_IMPORTS = "pyflyby.missing_imports"
+FORMATTING_IMPORTS = "pyflyby.format_imports"
+
+pyflyby_comm_targets= [MISSING_IMPORTS, FORMATTING_IMPORTS]
+
+# A map of the comms opened with a given target name.
+comms = {}
+
+# TODO: Document the expected contract for the different
+# custom comm messages
+
+def _register_target(target_name):
+    try:
+        ip = get_ipython()
+    except:
+        logger.debug("get_ipython() doesn't work")
+    else:
+        comm_manager = ip.kernel.comm_manager
```

Comment:
make sure this doesn't crash on older versions of ipython

Reply:
Is there a minimum target of IPython versions we intend to support?

Reply:
Any version you pick is fine.  Just make sure that for older versions it degrades gracefully (e.g. just doesn't have the new functionality, rather than breaking anything)

Reply:
Tested that pyflyby continues to work in older versions of ipython - even up to 0.13. For pyflyby in lab/notebook, tested that it works upto ipython==4.x. I don't think it is possible to use the notebook or lab with versions of ipython before 4.x (that was when ipython split up into multiple jupyter repos)

Reply:
It also breaks in python3 (later versions of ipython), but I guess that is expected now.

Reply:
I am adding Python 3 support in https://github.com/pyflyby/pyflyby/pull/10. I have only tested it against the latest version of IPython (7.5), but I suspect it works with all versions (getting the test suite itself to work against all IPython versions is challenging).

Reply:
It breaks [here](https://github.com/pyflyby/pyflyby/blob/64138ef593a7005b5977eff071976b480db8cbd8/lib/python/pyflyby/_dbg.py#L839)